__tests__/13_class_spec.ts

```diff
@@ -1,38 +1,102 @@
-import { createProxy, isChanged, markToTrack } from '../src/index';
+import {
+  affectedToPathList,
+  createProxy,
+  isChanged,
+  markToTrack,
+} from '../src/index';
 
 const noop = (_arg: unknown) => {
   // do nothing
 };
 
 describe('class spec', () => {
-  class C {
-    a = 1;
+  class Counter {
+    count = 1;
+
+    get doubled() { return this.count; }
+
+    set doubled(v) { this.count = v / 2; }
   }
 
   it('normal', () => {
     const proxyCache = new WeakMap();
-    const s1 = new C();
+    const s1 = new Counter();
     markToTrack(s1);
     const a1 = new WeakMap();
     const p1 = createProxy(s1, a1, proxyCache);
-    noop(p1.a);
+    noop(p1.count);
     expect(isChanged(s1, s1, a1)).toBe(false);
-    expect(isChanged(s1, { a: 1 }, a1)).toBe(false);
-    expect(isChanged(s1, { a: 2 }, a1)).toBe(true);
-    expect(Object.getPrototypeOf(p1) === C.prototype).toBe(true);
+    expect(isChanged(s1, { count: 1 }, a1)).toBe(false);
+    expect(isChanged(s1, { count: 2 }, a1)).toBe(true);
+    expect(Object.getPrototypeOf(p1) === Counter.prototype).toBe(true);
   });
 
   it('frozen', () => {
     const proxyCache = new WeakMap();
-    const s1 = new C();
+    const s1 = new Counter();
     Object.freeze(s1);
     markToTrack(s1);
     const a1 = new WeakMap();
     const p1 = createProxy(s1, a1, proxyCache);
-    noop(p1.a);
+    noop(p1.count);
     expect(isChanged(s1, s1, a1)).toBe(false);
-    expect(isChanged(s1, { a: 1 }, a1)).toBe(false);
-    expect(isChanged(s1, { a: 2 }, a1)).toBe(true);
-    expect(Object.getPrototypeOf(p1) === C.prototype).toBe(true);
+    expect(isChanged(s1, { count: 1 }, a1)).toBe(false);
+    expect(isChanged(s1, { count: 2 }, a1)).toBe(true);
+    expect(Object.getPrototypeOf(p1) === Counter.prototype).toBe(true);
+  });
+
+  it('object getters & setters', () => {
+    const obj = {
+      count: 1,
+      get doubled() { return this.count; },
+      set doubled(v) { this.count = v / 2; },
+    };
+    const proxyCache = new WeakMap();
+    const a1 = new WeakMap();
+    const p1 = createProxy(obj, a1, proxyCache);
+
+    // proxy observers the getter call
+    noop(p1.doubled);
+    expect(affectedToPathList(p1, a1)).toEqual([['doubled']]);
+
+    // setting proxy affects obj
+    p1.doubled = 4;
+    expect(obj.count).toBe(2);
```

Comment:
W/o the change, `p1.doubled = 4` fails b/c the setter is stripped as part of the unfreezing process:

```
  if (frozen) {
    handler.set = handler.deleteProperty = () => false;
  }
```

Which if I comment that out to "just see what happens", then `p1.doubled = 4` does not blow up, _but_ the write goes to the copy and not `obj`.

So, fixing the "is frozen detection" to be more accurate seems like the best fix + avoids both of these issues.